### PR TITLE
Include CopyButton for both token and XCH values of generate offers to enhance user experience

### PR DESCRIPTION
**Pull request by johnathon-b**
**User Story:**
As a user creating manual offer files, I want the ability to quickly and easily copy token and XCH values, so that I can streamline the process of generating offers and reduce the risk of errors.

**Description:**
In the current process of creating manual offer files, users have to manually select and copy the token, XCH and fee values, which can be time-consuming and prone to errors. To enhance the user experience and increase efficiency, we will utilize 'CopyButton' within the UI for token, XCH and fee values in the generate offers component.

### CODE REVIEW

File: src/components/trade_components/GenerateOffer.tsx [hunk 1]
```diff
@@ -446,7 +446,7 @@ const GenerateOffer: React.FC<GenerateOfferProps> = ({ data, devFee, setGenerate
                     
                     <p className="py-4 px-4 font-medium mb-12 bg-brandDark/10 rounded-xl">
                         <span>Suggested fee</span>
-                        <span className="font-normal pl-2 animate-fadeIn">{fee} {process.env.NEXT_PUBLIC_XCH}</span>
+                        <span><CopyButton copyText={fee}>{fee} {process.env.NEXT_PUBLIC_XCH}</CopyButton></span>
```
Comment (Yakuhito) [inline src/components/trade_components/GenerateOffer.tsx]: Why are the `<span>` classes removed here?

File: src/components/trade_components/GenerateOffer.tsx [hunk 2]
```diff
@@ -284,8 +284,8 @@ const GenerateOffer: React.FC<GenerateOfferProps> = ({ data, devFee, setGenerate
                         {/* If swap, add dev fee on top of quote */}
                         <div className="flex gap-2 items-center">
                             <Image src={e[0].image_url} width={30} height={30} alt="Token logo" className="rounded-full outline-brandDark/20 p-0.5" />
-                            <p>{amountWithFee(e)}</p>
-                            <p>{process.env.NEXT_PUBLIC_XCH === "TXCH" && e[0].name === "Chia" ? "Testnet Chia" : e[0].name === "Pair Liquidity Token" ? e[0].short_name : e[0].name}</p>
+                            <CopyButton copyText={amountWithFee(e)}>{amountWithFee(e)} {process.env.NEXT_PUBLIC_XCH === "TXCH" && e[0].name === "Chia" ? "Testnet Chia" : e[0].name === "Pair Liquidity Token" ? e[0].short_name : e[0].name}</CopyButton>
```
Comment (Yakuhito) [inline src/components/trade_components/GenerateOffer.tsx]: copyText should be a string, `amountWithFee(e)` is a number (that's why Vercel builds are failing - you can try running `npm build' locally to see the error)

File: src/components/trade_components/GenerateOffer.tsx (hunk 1 above)
Comment (Yakuhito) [reply src/components/trade_components/GenerateOffer.tsx]: also, `fee` is a number (needs to be converted to string)

**Review by Yakuhito — CHANGES_REQUESTED** [review]
Hey there! Thanks for your contribution - although it's easier to interact with the Chia wallet through WalletConnect, I think that the 'paste offer' functionality gives users more flexibility (e.g., they can use a completely different wallet - it only needs to support offers!). I left some comments about things that will need to be resolved before merging :)

**Comment by JonathanLangton1** [comment]
Hi, my thoughts on this are that instead of another button, the user should have the ability to click directly on the value (text) itself to copy. You can probably use the CopyButton component with additional class names to remove all styling.

Happy to make these changes if you'd like me to.

**Comment by johnathon-b** [comment]
I can change the styling today and push a commit. Should we still include the copy symbol so the user is aware of the functionality or make it something discoverable. 🤔

**Comment by JonathanLangton1** [comment]
> I can change the styling today and push a commit. Should we still include the copy symbol so the user is aware of the functionality or make it something discoverable. 🤔

I was thinking, for now, if you hover over the value text, just make the opacity of it 80% with cursor: pointer

**Comment by JonathanLangton1** [comment]
Should be good to go now 🙂

**Pull request merged**